modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -349,7 +349,7 @@ def concat(self, axis, other, **kwargs):
         if not isinstance(other, list):
             other = [other]
         assert all(
-            isinstance(o, type(self)) for o in other
+            isinstance(o, BaseQueryCompiler) for o in other
```

Comment:
This change suggests that we can concatenate different kinds of query compilers (`PandasQueryCompiler`, `SmallQueryCompiler`), but I don't see that being handled (possible I'm missing something!). Should this line be changed in this PR?